test/runtest.py

```diff
@@ -1495,19 +1495,87 @@ def h():
 
         ray.worker.cleanup()
 
-    def testInfiniteCustomResource(self):
-        # Make sure that -1 corresponds to an infinite resource capacity.
-        ray.init(num_custom_resource=-1)
+    def testTwoCustomResources(self):
+        ray.worker._init(
+            start_ray_local=True,
+            num_local_schedulers=2,
+            num_cpus=[3, 3],
+            resources=[{"CustomResource1": 1, "CustomResource2": 2},
+                       {"CustomResource1": 3, "CustomResource2": 4}])
+
+        @ray.remote(resources={"CustomResource1": 1})
+        def f():
+            time.sleep(0.001)
+            return ray.worker.global_worker.plasma_client.store_socket_name
+
+        @ray.remote(resources={"CustomResource2": 1})
+        def g():
+            time.sleep(0.001)
+            return ray.worker.global_worker.plasma_client.store_socket_name
+
+        @ray.remote(resources={"CustomResource1": 1, "CustomResource2": 3})
+        def h():
+            time.sleep(0.001)
+            return ray.worker.global_worker.plasma_client.store_socket_name
+
+        @ray.remote(resources={"CustomResource1": 4})
+        def j():
+            time.sleep(0.001)
+            return ray.worker.global_worker.plasma_client.store_socket_name
+
+        @ray.remote(resources={"CustomResource3": 1})
+        def k():
+            time.sleep(0.001)
+            return ray.worker.global_worker.plasma_client.store_socket_name
+
+        # The f and g tasks should be scheduled on both local schedulers.
+        self.assertEqual(len(set(ray.get([f.remote() for _ in range(50)]))), 2)
+        self.assertEqual(len(set(ray.get([g.remote() for _ in range(50)]))), 2)
+
+        local_plasma = ray.worker.global_worker.plasma_client.store_socket_name
+
+        # The h tasks should be scheduled only on the second local scheduler.
+        local_scheduler_ids = set(ray.get([h.remote() for _ in range(50)]))
+        self.assertEqual(len(local_scheduler_ids), 1)
+        self.assertNotEqual(list(local_scheduler_ids)[0], local_plasma)
+
+        # Make sure that tasks with unsatisfied custom resource requirements do
+        # not get scheduled.
+        ready_ids, remaining_ids = ray.wait([j.remote(), k.remote()],
+                                            timeout=500)
+        self.assertEqual(ready_ids, [])
+
+        ray.worker.cleanup()
+
+    def testManyCustomResources(self):
+        num_custom_resources = 10000
```

Comment:
Note that if you make this number very large (e.g., 10x bigger), the command that we use to start the local scheduler from the command line is extremely long and will fail.